packages/core/src/config/parse.ts

```diff
@@ -1191,30 +1230,25 @@ permission to call the 'upgradeTo' function on each of them.
           openzeppelinStorageLayouts
         )
 
-        const { input, output } = readBuildInfo(
-          artifactPaths[referenceName].buildInfoPath
-        )
-        const newStorageLayout = getOpenZeppelinStorageLayout(
-          contractConfig.contract,
-          input,
-          output,
-          contractConfig.proxyType
-        )
-
         assertStorageCompatiblePreserveKeywords(
           contractConfig,
           previousStorageLayout,
           newStorageLayout
         )
 
-        // We could check for the `skipStorageCheck` in the outer for-loop, but this makes it easy to
-        // support more granular storage layout config options in the future.
-        if (parsedConfig.options.skipStorageCheck !== true) {
-          // Run OpenZeppelin's storage slot checker.
+        if (
+          // If the user has disabled storage checks for all contracts
+          parsedConfig.options.skipStorageCheck !== true &&
```

Comment:
I think we should remove `skipStorageCheck` from `options`. It doesn't really make sense for it to be the only OpenZeppelin option that's on the granularity of the entire config, especially considering `unsafeAllowRenames` also operates on the storage layouts